frontend/src/metabase/dashboard/actions/save.js

```diff
@@ -156,6 +161,22 @@ export const saveDashboardAndCards = createThunkAction(
         }
       }
 
+      const pageTitleTemplateChange = getPageTitleTemplateChange(state);
+      if (dashboard.is_app_page && pageTitleTemplateChange) {
+        const dataAppId = Number(params.slug);
+        const dataApp = DataApps.selectors.getObject(state, {
+          entityId: dataAppId,
+        });
+        await dispatch(
+          updateDataAppPageTitle(
+            dataApp,
+            dashboard.id,
+            pageTitleTemplateChange,
+          ),
+        );
+        dispatch(setPageTitleTemplate(null));
```

Comment:
I think this has the weird behavior that if we edit a page title, then "cancel" editing, the page title edit persists locally, until we refresh the page, then the edit is lost.